src/test/java/io/stargate/sgv2/jsonapi/service/processor/MeteredCommandProcessorTest.java

```diff
@@ -0,0 +1,131 @@
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.quarkus.test.junit.mockito.InjectMock;
+import io.smallrye.mutiny.Uni;
+import io.stargate.sgv2.api.common.StargateRequestInfo;
+import io.stargate.sgv2.common.testprofiles.NoGlobalResourcesTestProfile;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandContext;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandResult;
+import io.stargate.sgv2.jsonapi.api.model.command.impl.CountDocumentsCommands;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+@QuarkusTest
+@TestProfile(NoGlobalResourcesTestProfile.Impl.class)
+public class MeteredCommandProcessorTest {
+  @Inject MeteredCommandProcessor meteredCommandProcessor;
+  @InjectMock protected CommandProcessor commandProcessor;
+  @InjectMock protected StargateRequestInfo stargateRequestInfo;
+  @Inject ObjectMapper objectMapper;
+
+  @Nested
+  class CustomMetrics {
+    @Test
+    public void metrics() throws Exception {
+      String json =
+          """
+          {
+            "countDocuments": {
+
+            }
+          }
+          """;
+
+      CountDocumentsCommands countCommand =
+          objectMapper.readValue(json, CountDocumentsCommands.class);
+      CommandContext commandContext = new CommandContext("namespace", "collection");
+      CommandResult commandResult = new CommandResult(Collections.emptyList());
+      Mockito.when(commandProcessor.processCommand(commandContext, countCommand))
+          .thenReturn(Uni.createFrom().item(commandResult));
+      Mockito.when(stargateRequestInfo.getTenantId()).thenReturn(Optional.of("test-tenant"));
+      meteredCommandProcessor.processCommand(commandContext, countCommand).await().indefinitely();
+      String metrics = given().when().get("/metrics").then().statusCode(200).extract().asString();
+      List<String> httpMetrics =
+          metrics
+              .lines()
+              .filter(
+                  line ->
+                      line.startsWith("command_processor_process")
+                          && line.contains("error=\"false\""))
+              .toList();
+
+      assertThat(httpMetrics)
+          .satisfies(
+              lines -> {
+                assertThat(lines.size()).isEqualTo(3);
+                lines.forEach(
+                    line -> {
+                      assertThat(line).contains("command=\"CountDocumentsCommands\"");
+                      assertThat(line).contains("tenant=\"test-tenant\"");
+                      assertThat(line).contains("error=\"false\"");
+                      assertThat(line).contains("error_code=\"NA\"");
+                      assertThat(line).contains("error_class=\"NA\"");
```

Comment:
should this be `none` and not `NA`? because on happy path we should not set to NA, as we set when there is an error, but there's no error code?.. Not sure though..

Reply:
As discussed keeping this as NA